# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tibit/hexedit/00-hexedit-vi-keys.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -u hexedit/interact.c hexedit-hacked/interact.c
--- hexedit/interact.c	Wed Jun 19 13:55:40 2002
+++ hexedit-hacked/interact.c	Wed Oct  9 12:04:18 2002
@@ -360,6 +360,39 @@
       next_line();
       break;
 
+      /* VI keys */
+    case 'h': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      backward_char();
+      break;
+
+    case 'l': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      forward_char();
+      break;
+
+    case 'k': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      previous_line();
+      break;
+
+    case 'j': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      next_line();
+      break;
+
     case KEY_UP:
     case CTRL('P'):
       previous_line();
@@ -443,6 +476,14 @@
 
     case CTRL('G'):
     case KEY_F(4):
+      goto_char();
+      break;
+
+    case 'g':
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
       goto_char();
       break;
 
